files/fuzzers/shuffle/fuzzer.cpp

```diff
@@ -67,12 +70,52 @@ class Lighthouse : public Rust {
     return ret;
   }
 };
+
+class Nimbus : public Nim {
+  std::optional<std::vector<uint8_t>> run(
+      const std::vector<uint8_t> &data) override {
+    uint16_t count;
+    // TODO(gnattishness) use c new instead of malloc?
+    // any reason not to have this point to the existing vector.data?
+    // no need to malloc right?
+    uint8_t *seed = reinterpret_cast<uint8_t *>(malloc(32));
+
+    if (data.size() < sizeof(count) + 32) {
+      return std::nullopt;
+    }
+
+    memcpy(&count, data.data(), sizeof(count));
+    count %= 100;
+    memcpy(seed, data.data() + sizeof(count), 32);
+
+    std::vector<uint64_t> input(count);
+
+    /* input[0..count] = 0..count */
```

Comment:
For `nfuzz_shuffle` you don't have to provide this input, it is purely used as an output. It currently does expect that this vector is zeroed though (due to some issue on our side), so I think this might be the problem for the test to fail immediately.

Reply:
Much thanks!